test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -354,4 +361,75 @@ private String errorMessage(ExecutableSection executableSection, Throwable t) {
     protected boolean randomizeContentType() {
         return true;
     }
+
+
+    /**
+     * Sniff the cluster for host metadata if it hasn't already been sniffed. This isn't the
+     * same thing as using the {@link Sniffer} because:
+     * <ul>
+     * <li>It doesn't replace the hosts that that {@link #client} communicates with
+     * <li>If there is already host metadata it skips running. This behavior isn't
+     *     thread safe but it doesn't have to be for our tests.
+     * </ul>
+     */
+    private void sniffHostMetadata(RestClient client) throws IOException {
+        Node[] nodes = client.getNodes();
+        boolean allHaveRoles = true;
+        for (Node node : nodes) {
+            if (node.getRoles() == null) {
+                allHaveRoles = false;
+                break;
+            }
+        }
+        if (allHaveRoles) {
+            // We already have resolved metadata.
+            return;
+        }
+        // No resolver, sniff one time and resolve metadata against the results
+        ElasticsearchHostsSniffer.Scheme scheme =
+            ElasticsearchHostsSniffer.Scheme.valueOf(getProtocol().toUpperCase(Locale.ROOT));
+        /*
+         * We don't want to change the list of nodes that the client communicates with
+         * because that'd just be rude. So instead we replace the nodes with nodes the
+         * that
+         */
+        ElasticsearchHostsSniffer sniffer = new ElasticsearchHostsSniffer(
+                adminClient(), ElasticsearchHostsSniffer.DEFAULT_SNIFF_REQUEST_TIMEOUT, scheme);
+        attachSniffedMetadataOnClient(client, nodes, sniffer.sniffHosts());
+    }
+
+    static void attachSniffedMetadataOnClient(RestClient client, Node[] originalNodes, List<Node> nodesWithMetadata) {
+        Set<HttpHost> originalHosts = Arrays.stream(originalNodes)
+                .map(Node::getHost)
+                .collect(Collectors.toSet());
+        List<Node> sniffed = new ArrayList<>();
+        for (Node node : nodesWithMetadata) {
+            /*
+             * getHost is the publish_address of the node which, sometimes, is
+             * ipv6 and, sometimes, our original address for the node is ipv4.
+             * In that case the ipv4 address should be in getBoundHosts. If it
+             * isn't then we'll end up without the right number of hosts which
+             * will fail down below with a pretty error message.
+             */
```

Comment:
I think I see now why the `boundHosts` matter. This is quite annoying though as we need it only because of how we want to match existing nodes with what we find in the cluster state. I guess that replacing the nodes that we found would be problematic?

Reply:
I'm not 100% sure to be honest. It *feels* like we should honor the list of hosts but that it'll rarely matter. It just feel super weird to replace the hosts.

Reply:
And how about activating ordinary sniffing instead , maybe with a very high resniff interval or something like that? I do see why replacing the hosts is not ideal, but I think that not doing it complicates things in our production code, which is even worse.